js/classes/game_engine.js

```diff
@@ -74,31 +94,46 @@ class GameEngine {
      * This method is going to go through all entities and allow them to update their position.
      */
     update() {
-        // Update the HUD and Crosshair regardless of whether the game is running or not
-        HUD.update();
-        CROSSHAIR.update();
-
         if (this.running) {
-            // Iterate forward through the entities array. Update everything.
-            for (let i = 0; i < this.entities.length; i++) {
-                let entity = this.entities[i];
-                if (!entity.removeFromWorld) {
+            // (1) Update the background entities:
+            this.entities.background.forEach((entity) => {
+                if (entity.removeFromWorld) {
+                    const i = this.entities.background.indexOf(entity);
+                    this.entities.background.splice(i, 1);
+                } else {
```

Comment:
I see why you refactored it this way but the only problem is that it allows entities to hang around for an extra update after setting the removeFromWorld flag. Not likely to matter but I thought I'd mention it.

Reply:
I don't think I changed anything Order-of-Operations wise for removal. The removeFromWorld flag is set, and then its next update method it will be removed. Actually removing an entity immediately is much more verbose, and not really using the code wisely IMO.